topic-operator/src/main/java/io/strimzi/operator/topic/v2/Either.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic.v2;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+class Either<X, Y> {
```

Comment:
A JavaDoc might be great to explain what it does / why it exists.

Reply:
@fvaleri I didn't know that convention. I can refactor it, but I'll do that after fixing other review comments since it might confuse github if there are any comments at call sites.

@scholzj I'll add some Javadoc. I'll also rename the `X` and `Y` to `L` and `R` to better reflect their relationship to `left` and `right`.

FWIW, I considered implementing this as a `sealed interface` with two permitted `final` implementations, but decided the `boolean left` was about as good.

@see-quick did you have any other comments/suggestions on this, since you mentioned it in a private chat?